pulsar-common/src/main/java/org/apache/pulsar/common/policies/data/EnsemblePlacementPolicyConfig.java

```diff
@@ -30,6 +31,8 @@
     private final Class policyClass;
     private final Map<String, Object> properties;
 
+    private static ObjectMapper mapper = new ObjectMapper();
```

Comment:
This is not thread safe and also is not used the preferred configuration for Jackson. `pulsar-common` has already `ObjectMapperFactory.getThreadLocal()` which is used for this.

Reply:
Thanks for the reminder. I change to use the `ObjectMapperFactory.getThreadLocal()`. Please take a look again.

Reply:
@merlimat I am not sure I get your point.
AFAIK it is very common to have static fields holding ObjectMapper if you do not need to change the configuration.
Usually you create one ObjectMapper per class or per group of classes (internally it holds a cache)

https://stackoverflow.com/questions/3907929/should-i-declare-jacksons-objectmapper-as-a-static-field

Reply:
btw using consistently ObjectMapperFactory.getThreadLocal() is good to me

Reply:
We've seen concurrency issues in the past resulting in random exceptions. If I'm not remembering wrong, they were related to the caching of the classes introspection in the object mapper.

Reply:
@merlimat  that makes sense to me.
In fact usually you create a `static` ObjectMapper when you are sure that it will deal only with a limited and fixed set classes .

in this case the ObjectMapper will deal only with this class, and thus it is safe to use a static instance.,

As just said before, I prefer to have a consistent way of coding, so here in Pulsar is it fine to me to use ObjectManagerFactory. The overhead is to create a ObjectMapper (and the internal cache) per thread instead of having just one per target class.